File: code_challenge_solution/javascript/144 Password & passphrase generator.js
```javascript
function generatePassword(length) {
  const lowercase = 'abcdefghijklmnopqrstuvwxyz';
  const uppercase = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
  const numbers = '0123456789';
  const specialChars = '!@#$%^&*()-_=+[]{}<>,.?/~`|';

  let characters = lowercase + uppercase + numbers + specialChars;

  let password = '';
  for (let i = 0; i < length; i++) {
    let randomIndex = Math.floor(Math.random() * characters.length);
    password += characters.charAt(randomIndex);
  }

  return password;
}

function generatePassphrase(words) {
  const dictionary = [
    'apple', 'banana', 'cat', 'dog', 'elephant', 'fish', 'gorilla', 'horse',
    'igloo', 'jellyfish', 'kangaroo', 'lemon', 'monkey', 'narwhal', 'octopus',
    'penguin', 'quail', 'rabbit', 'squirrel', 'tiger', 'unicorn', 'vulture',
    'walrus', 'xylophone', 'yak', 'zebra'
  ];

  let passphrase = '';
  for (let i = 0; i < words; i++) {
    let randomIndex = Math.floor(Math.random() * dictionary.length);
    passphrase += dictionary[randomIndex] + ' ';
  }

  return passphrase.trim();
}

// Usage:
console.log('Generated Password:', generatePassword(10));
console.log('Generated Passphrase:', generatePassphrase(4));
```
Note: The above code generates a password by combining lowercase, uppercase, numbers, and special characters. The generated password has a specified length. The code also generates a passphrase by randomly selecting words from a dictionary. The number of words in the passphrase can be specified.

File: code_challenge_solution/java/144 Password & passphrase generator.java
import java.util.Random;

public class PasswordGenerator {

    private static final String CHAR_LOWER = "abcdefghijklmnopqrstuvwxyz";
    private static final String CHAR_UPPER = CHAR_LOWER.toUpperCase();
    private static final String NUMBER = "0123456789";
    private static final String SPECIAL_CHARS = "!@#$%&*";

    private static final int PASSWORD_LENGTH = 10;

    private static Random random = new Random();

    public static void main(String[] args) {

        System.out.println("Secure Password Generator");
        System.out.println("-------------------------");
        System.out.println();

        System.out.println("Generated Password: " + generatePassword());
        System.out.println("Generated Passphrase: " + generatePassphrase());
    }

    private static String generatePassword() {
        String characters = CHAR_LOWER + CHAR_UPPER + NUMBER + SPECIAL_CHARS;
        StringBuilder password = new StringBuilder(PASSWORD_LENGTH);
        for (int i = 0; i < PASSWORD_LENGTH; i++) {
            int randomIndex = random.nextInt(characters.length());
            password.append(characters.charAt(randomIndex));
        }
        return password.toString();
    }

    private static String generatePassphrase() {
        String[] words = {"apple", "banana", "cherry", "date", "elderberry", "fig", "grape", "honeydew", "kiwi", "lemon"};
        StringBuilder passphrase = new StringBuilder(PASSWORD_LENGTH);
        for (int i = 0; i < PASSWORD_LENGTH; i++) {
            int randomIndex = random.nextInt(words.length);
            passphrase.append(words[randomIndex]).append(" ");
        }
        return passphrase.toString().trim();
    }
}